src/EditorFeatures/Core/Implementation/AutomaticCompletion/BraceCompletionSessionProvider.BraceCompletionSession.cs

```diff
@@ -115,7 +123,14 @@ private void Start(CancellationToken cancellationToken)
                 // switch the closing point from positive to negative tracking so that the closing point stays against the closing brace
                 ClosingPoint = SubjectBuffer.CurrentSnapshot.CreateTrackingPoint(ClosingPoint.GetPoint(SubjectBuffer.CurrentSnapshot), PointTrackingMode.Negative);
 
-                var changesAfterStart = _session.GetChangesAfterCompletion(this, cancellationToken);
+                var contextAfterStart = GetBraceCompletionContext();
+                if (context == null)
+                {
+                    caretPreservingTransaction.Complete();
+                    return;
+                }
```

Comment:
this flow is very weird.  i would prefer a single place where we complete the transaction (since all paths complete here).  but have the special cases code if you `ApplyBraceCompletionResult`